Source/options.h

```diff
@@ -0,0 +1,230 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include "all.h"
+#include "SDL_stdinc.h"
+#include "../3rdParty/Storm/Source/storm.h"
+
+DEVILUTION_BEGIN_NAMESPACE
+
+/**
+ * @brief Base non-templated class for all options. Provides saving and loading functionality and a way to group every
+ * type of option together.
+*/
+class OptionBase {
+public:
+	/**
+	 * @brief Loads data from the ini into this instance.
+	 * @param sectionName The name of the section that contains the data for this option.
+	*/
+	virtual void Load(const std::string& sectionName) = 0;

```

Comment:
Perhaps section name should be part of the option itself, instead of being passed into these methods? We don't expect to save the same option in different sections.

Reply:
This allows me to pass the section name from the parent. I disagree that the setting itself should know it's section, since the setting is inside a container object which itself is mapped to a section.

Right now, if we move a setting from one section to the other (from one `OptionGroup` class to another) it will automatically pick the section from that parent.

If we allow the setting itself to have its own section name, we open the door for inconsistencies like a setting being inside a parent class, but being part of a section for another class. That would be less than ideal.

Reply:
OK